check/check_test.go

```diff
@@ -192,3 +193,70 @@ func TestGetFirstValidSubCheck(t *testing.T) {
 		}
 	}
 }
+func TestTextToCommand(t *testing.T) {
+	type TestCase struct {
+		auditCommand          string
+		expectedSlicedCommand []string
+	}
+
+	// For each test case, we want get the different commands separated by | and exclude the \\|
+	// which stand for the use of | as OR operator and not as a pipe.
+	testCases := []TestCase{
+		{
+			// Test for mixed case of pipeline and or cases
+			auditCommand: "grep -E -i \"(\\\\v\\|\\\\r\\|\\\\m\\|\\\\s)\\|$(grep '^ID=' /etc/os-release | cut -d= -f2 | sed -e 's/\"//g'))\" /etc/motd",
+			expectedSlicedCommand: []string{
+				"grep -E -i \"(\\\\v|\\\\r|\\\\m|\\\\s)|$(grep '^ID=' /etc/os-release",
+				"cut -d= -f2",
+				"sed -e 's/\"//g'))\" /etc/motd"},
```

Comment:
I think this test is correct in the sense that given this input to `testToCommand` I agree this is what we'd expect the output to be from that function. 

But, if the command gets split up like this, I don't see how it can actually run. Here's why:

The benchmark says run the following command: 
```
grep -E -i "(\\v|\\r|\\m|\\s|$(grep '^ID=' /etc/os-release | cut -d= -f2 | sed -e 's/"//g'))" /etc/motd
```

The $(...) part means we need to execute that part and use the results. So we could write the same command like this: 

```
$ RESULT = $(grep '^ID=' /etc/os-release | cut -d= -f2 | sed -e 's/"//g')
$ echo $RESULT
ubuntu
$ grep -E -i "(\\v|\\r|\\m|\\s|$RESULT)" /etc/motd
```
The way the commands are currently split up, the first one attempts to run 

```
grep -E -i \"(\\v|\\r|\\m|\\s)|$(grep '^ID=' /etc/os-release"
```
which results in 

```
$ grep -E -i \"(\\v|\\r|\\m|\\s)|$(grep '^ID=' /etc/os-release"
-bash: syntax error near unexpected token `('
```

What I eventually realised is that we do all this work to split up the commands _and then we just run the original audit command anyway!_ See my comment elsewhere, and PR https://github.com/aquasecurity/bench-common/pull/89

Reply:
Yes, all of this is meant to check if the commands are valid (ignoring the parameters), the part that really gets checked afterward is cmd.Path, which will be just /bin/grep, /bin/cat, /bin/sed.
As you mentioned after this validation audit will execute as is.